vtex-apps/product-price
# Feature/installments list

#### What problem is this solving?

Add block `product-installments.list`

#### How to test it?

1. [Workspace](https://installmentslist--storecomponents.myvtex.com/vintage-phone/p)
2. Click on *See all possible installments*

You can edit the message in the site-editor as well:

[Workspace](https://installmentslist--storecomponents.myvtex.com/admin/cms/site-editor/vintage-phone/p)
![image](https://user-images.githubusercontent.com/8517023/88198039-a56f3980-cc19-11ea-8a7a-08c2bf1cd12b.png)

#### Depends on

1. https://github.com/vtex-apps/store-resources/pull/128

#### Code used in the `store-theme`

This code was used just to create the modal, but you can use the component by just adding `product-installments.list` anywhere in the product page

```jsonc
// price.jsonc

  "modal-trigger#installments-list": {
    "children": [
      "rich-text#installments-list",
      "modal-layout#installments-list"
    ]
  },
  "rich-text#installments-list": {
    "props": {
      "text": "See all possible installments",
      "blockClass": "installmentsList"
    }
  },
  "modal-layout#installments-list": {
    "props": {
      "blockClass": "installmentsList"
    },
    "children": [
      "modal-header#installments-list",
      "modal-content#installments-list",
      "modal-actions#installments-list"
    ]
  },
  "modal-header#installments-list": {
    "props": {
      "showCloseButton": false
    },
    "children": [
      "rich-text#modal-installments-header"
    ]
  },
  "rich-text#modal-installments-header": {
    "props": {
      "text": "#### Installments List",
      "blockClass": "installmentsHeader"
    }
  },
  "modal-content#installments-list": {
    "children": [
      "product-installments-list#installments-modal"
    ]
  },
  "product-installments-list#installments-modal": {
    "children": [
      "product-installments-list-item#installments-modal"
    ]
  },
  "product-installments-list-item#installments-modal": {
    "props": {
      "message": "Up to {installmentsNumber}x {installmentValue}{hasInterest, select, true {, {interestRate} of interest} false { interest-free}} in {paymentSystemName}"
    }
  },
  "modal-actions#installments-list": {
    "props": {
      "blockClass": "installmentsList"
    },
    "children": [
      "modal-actions.close#installments-list"
    ]
  },
  "modal-actions.close#installments-list": {
    "props": {
      "label": "Close"
    }
  }
```

#### Screenshots or example usage:

![image](https://user-images.githubusercontent.com/8517023/88197364-c2573d00-cc18-11ea-8f1c-12a680310f1c.png)




#### How does this PR make you feel? [:link:](http://giphy.com/)



![](put .gif link here - can be found under "advanced" on giphy)
